Add create/join/start game socket (server side)

lhossackucalgary opened the pull request.

lhossackucalgary committed:
new game + join game socket

lhossackucalgary committed:
Merge branch 'master' into feature/game-socket

lhossackucalgary committed:
add game/startGame & game/actionFailed

lhossackucalgary committed:
fix broadcast to game

lhossackucalgary committed:
rm dbg messages

jackadrianglass reviewed (approved):
Looks good to me. I didn't find anything that needed changing

lhossackucalgary committed:
Implement socket events for turns 

Start & end turn events 
- includes automatic ending of turn from server side after 3 minutes
- UTC end of turn time sent to all clients in game
- player order is shuffled at start of game
- all active games for a user is sent on socket connect (1 at a time) - 
same method as when a new game is started (all occurences can just 
append to active game list)

lhossackucalgary committed:
Vuex updates for game management

- create, 
- active, 
- joined, 
- actionFailed

Does NOT yet include turn mgmt or game object on client side

lhossackucalgary committed:
repair navbar

use objects instead of maps for vue reactivity*

lhossackucalgary committed:
Merge branch 'master' into feature/game-socket

jackadrianglass reviewed (commented):
Still looks good to me
